src/main/java/com/autotune/experimentManager/data/EMMapper.java

```diff
@@ -1,16 +1,23 @@
 package com.autotune.experimentManager.data;
 
+import com.autotune.common.experiments.ExperimentTrial;
+
+import java.util.HashMap;
 import java.util.LinkedList;
+import java.util.List;
 import java.util.concurrent.ConcurrentHashMap;
 
 public class EMMapper {
     private static volatile EMMapper emMapper = null;
     private volatile ConcurrentHashMap<String, ExperimentTrialData> emMap = null;
     private volatile ConcurrentHashMap<String, LinkedList<String>> emDeploymentRunIdMap = null;
 
+    private volatile ConcurrentHashMap<String, HashMap<String, ExperimentTrial>> emExpTrialMap = null;
```

Comment:
Change it to

    private volatile ConcurrentHashMap<String,  ExperimentTrial> emExpTrialMap = null;

Reply:
I'm expecting that changing from a data structure mapping (string, hashmap) to data point mapping (string, object) makes autotune to lose the ability of storing the older trial information. Atleast till a point we have a DB to record older trial information we need to maintain it locally via global maps (which the current implementation does).